src/usePositionStickyObserver/usePositionStickyObserver.ts

```diff
@@ -0,0 +1,58 @@
+import { Ref, computed, onMounted, ref } from 'vue'
+import { MaybeRefOrGetter } from '@/types/maybe'
+import { useIntersectionObserver } from '@/useIntersectionObserver'
+import { toValue } from '@/utilities/vue'
+
+export type UsePositionStickyObserverResponse = {
+  stuck: Ref<boolean>,
+}
+
+export type UsePositionStickyObserverOptions = {
+  rootMargin?: string,
+  boundingElement?: MaybeRefOrGetter<HTMLElement | undefined>,
+}
+
+const usePositionStickyObserverDefaultOptions = {
+  rootMargin: '-1px 0px 0px 0px',
+  boundingElement: document.body,
+}
+
+export function usePositionStickyObserver(
+  element: MaybeRefOrGetter<HTMLElement | undefined>,
+  options?: MaybeRefOrGetter<UsePositionStickyObserverOptions>,
+): UsePositionStickyObserverResponse {
+  const elementRef = computed(() => toValue(element))
+  const optionsRef = computed(() => {
```

Comment:
Rather than creating this computed and then a second computed for the visibility observer, I'd just have this return the type the visibility observer expects. That way we don't need two computeds

Reply:
Really great calls. Updated! Will go ahead and merge to plow ahead, if you look back and catch anything else np.